modules/search-as-you-type/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```
@@ -0,0 +1,432 @@
+
+        @Override
+        public Mapper.Builder<?, ?> parse(String name,
+                                          Map<String, Object> node,
+                                          ParserContext parserContext) throws MapperParsingException {
+
+            final SearchAsYouTypeFieldMapper.Builder builder = new SearchAsYouTypeFieldMapper.Builder(name);
+
+            NamedAnalyzer analyzer = parserContext.getIndexAnalyzers().getDefaultIndexAnalyzer();
+            int maxShingleSize = Defaults.MAX_SHINGLE_SIZE;
+
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                final Map.Entry<String, Object> entry = iterator.next();
+                final String fieldName = entry.getKey();
+                final Object fieldNode = entry.getValue();
+                if (fieldName.equals("type")) {
+                    continue;
+                } else if (fieldName.equals("index_options")) {
+                    builder.indexOptions(nodeIndexOptionValue(fieldNode));
+                    iterator.remove();
+                } else if (fieldName.equals("fields")) {
+                    parseMultiField(builder, name, parserContext, fieldName, fieldNode);
+                    iterator.remove();
+                } else if (fieldName.equals("analyzer")) {
+                    final String analyzerName = fieldNode.toString();
+                    analyzer = parserContext.getIndexAnalyzers().get(analyzerName);
+                    if (analyzer == null) {
+                        throw new MapperParsingException("Analyzer [" + analyzerName + "] not found for field  [" + name + "]");
+                    }
+                    iterator.remove();
+                } else if (fieldName.equals("max_shingle_size")) {
+                    maxShingleSize = XContentMapValues.nodeIntegerValue(fieldNode);
+                    iterator.remove();
+                }
+            }
```

Should we throw an error if the map is not empty (has some fields that we don't support) ?

---

I assumed whatever calls this method did that afterwards, it looks like in at least some places they do. Should I throw one here too? https://github.com/elastic/elasticsearch/blob/3c03975ce396ff3e6f27b79d26e77a4cf1081f67/server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java#L276-L284